From c55b1f2b645ba561268d253ed801e15e26b644c5 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Tue, 3 Jul 2018 11:29:22 +0800
Subject: [PATCH 4146/5242] MLK-18609-8 usb: cdns3: gadget: do not support
 multiple requests for ep0

commit  4a4232458cc3eb9090ad7246e6806b2cd630bfd0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Current code does not support queuing multiple requests for ep0, return
-EOPNOTSUPP if that happens.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 57782a2..b3d81aa 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1429,6 +1429,13 @@ static int usb_ss_gadget_ep0_queue(struct usb_ep *ep,
 	}
 
 	spin_lock_irqsave(&usb_ss->lock, flags);
+	if (!list_empty(&usb_ss_ep->request_list)) {
+		dev_err(&usb_ss->dev,
+			"can't handle multiple requests for ep0\n");
+		spin_unlock_irqrestore(&usb_ss->lock, flags);
+		return -EOPNOTSUPP;
+	}
+
 	ret = usb_gadget_map_request_by_dev(usb_ss->sysdev, request,
 			usb_ss->ep0_data_dir);
 	if (ret) {
-- 
1.7.9.5

